plugins/scaffolder-backend/src/lib/catalog/CatalogEntityClient.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetch from 'node-fetch';
+import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
+import {
+  ConflictError,
+  NotFoundError,
+  PluginEndpointDiscovery,
+} from '@backstage/backend-common';
+
+/**
+ * A catalog client tailored for reading out entity data from the catalog.
+ */
+export class CatalogEntityClient {
+  private readonly discovery: PluginEndpointDiscovery;
+
+  constructor(options: { discovery: PluginEndpointDiscovery }) {
+    this.discovery = options.discovery;
+  }
+
+  /**
+   * Looks up a single template using a template name.
+   *
+   * Throws a NotFoundError or ConflictError if 0 or multiple templates are found.
+   */
+  async findTemplate(templateName: string): Promise<TemplateEntityV1alpha1> {
+    const params = new URLSearchParams();
+    params.append('kind', 'Template');
+
+    params.append('metadata.name', templateName);
```

Comment:
We'd need to filter on the correct `'metadata.namespace'` here too

Reply:
Although tbh that could definitely be a followup issue, as the scaffolder only handles the name atm

Reply:
Resolve it for now?

Reply:
Yeah let's do that improvement separately then